### PR TITLE
Feat/implement edit profile

### PR DESCRIPTION
**Pull request by AniaPiWo**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Requirements
The task requirements are as follows:
1. Clicking on "Save and preview profile" button in Edit profile form should submit form data to the database and redirect to My profile page.
2. Submitting form data should be sent to our API in the format declared in the required payload schema.
3. The Edit profile form should be available from the "/my-profile/edit" path.
4. When the submission is successful, it should redirect to the My profile page and display the saved values using the GET endpoint.

### Review

Based on the provided code changes, let's review if they cover the task requirements.

#### app/(profile)/my-profile/edit/page.module.scss
- No changes found.

#### app/(profile)/my-profile/edit/page.tsx
- The code changes include handling the form submission and redirecting to the My profile page.
- The form data is sent to the API endpoint for updating the profile.
- The payload schema is not explicitly defined in this file, but it is used to define the type of the form data.

#### app/(profile)/my-profile/page.tsx
- No changes found.

#### app/api/profiles/[id]/route.ts
- No changes found.

#### app/api/profiles/me/route.ts
- The code changes include handling the PUT request to update the profile.
- The endpoint matches the required endpoint for updating the profile.

#### css/Button.module.css
- No changes found.

#### css/Button.module.min.css
- No changes found.

#### css/Checkbox.module.css
- No changes found.

#### css/Checkbox.module.min.css
- No changes found.

#### css/Container.module.css
- No changes found.

#### css/Container.module.min.css
- No changes found.

#### css/CopyEmail.module.css
- No changes found.

#### css/CopyEmail.module.min.css
- No changes found.

#### css/CreateProfileTopBar.module.css
- No changes found.

#### css/CreateProfileTopBar.module.min.css
- No changes found.

#### css/DefaultHeader.module.css
- No changes found.

#### css/DefaultHeader.module.min.css
- No changes found.

#### css/DevTypeButton.module.css
- No changes found.

#### css/DevTypeButton.module.min.css
- No changes found.

#### css/DropdownBio.module.css
- No changes found.

#### css/DropdownBio.module.min.css
- No changes found.

#### css/DropdownFilter.module.css
- No changes found.

#### css/DropdownFilter.module.min.css
- No changes found.

#### css/Filters.module.css
- No changes found.

#### css/Filters.module.min.css
- No changes found.

#### css/GoBackButton.module.css
- No changes found.

#### css/GoBackButton.module.min.css
- No changes found.

#### css/LocationPreferences.module.css
- No changes found.

#### css/LocationPreferences.module.min.css
- No changes found.

#### css/LogOutBtn.module.css
- No changes found.

### Conclusion
Based on the provided code changes, it appears that the requirements have been met:
1. The form submission is handled and the form data is sent to the API endpoint for updating the profile.
2. The Edit profile form is available from the "/my-profile/edit" path.
3. Upon successful submission, the user is redirected to the My profile page and the saved values are displayed using the GET endpoint.

Please note that this review is based solely on the provided code changes and does not take into account any other parts of the application.

**Review by wawrzonnn — APPROVED** [review]

**Review by Sumick — CHANGES_REQUESTED** [review]
Looks good, few improvements to implement before merging 🙌

File: src/components/Headers/DefaultHeader/DefaultHeader.tsx [hunk 1]
```diff
@@ -23,13 +26,38 @@ const DefaultHeader = () => {
...
+                  height={38}
+                  alt="github avatar"
+                />
+              )}
+              <p className={styles.githubAccName}>{name}</p>
+            </div>
+            {profileId === null ? (
+              <Button
+                onClick={() => router.push('/create-profile')}
+                variant={'primary'}
```
Comment (wawrzonnn) [inline src/components/Headers/DefaultHeader/DefaultHeader.tsx:49]: you should use routes from src/utils/routes.ts

File: src/components/Headers/DefaultHeader/DefaultHeader.tsx [hunk 2]
```diff
@@ -23,13 +26,38 @@ const DefaultHeader = () => {
...
+              >
+                Create profile
+              </Button>
+            ) : (
+              <Button
+                onClick={() => router.push('/my-profile')}
+                variant={'primary'}
```
Comment (wawrzonnn) [inline src/components/Headers/DefaultHeader/DefaultHeader.tsx:56]: you should use routes from src/utils/routes.ts

File: src/services/edit-profile-form-service.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  bio: Yup.string().required('Bio is required'),
+  country: Yup.string().required('Country is required'),
+  city: Yup.string().required('City is required'),
+  openToRelocationCountry: Yup.boolean().oneOf([true, false]),
+  remoteOnly: Yup.boolean().oneOf([true, false], 'This field must be checked'),
+  position: Yup.string().required('Position is required'),
+  seniority: Yup.string().required('Seniority is required'),
+  techStack: Yup.string().required('Tech stack is required'),
+  // employment: Yup.array().of(Yup.string().oneOf(['FULL_TIME', 'PART_TIME', 'CONTRACT'])).min(1, 'Employment type is required'),
+})
```
Comment (wawrzonnn) [inline src/services/edit-profile-form-service.tsx:56]: we need to add validation here

**Review by wawrzonnn — APPROVED** [review]
good job, i left a few comments

File: src/services/edit-profile-form-service.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      isPublished: values.isPublished,
+    }
+    try {
+      await updateMyProfile(payload)
+      console.log('Profile edited successfully')
+    } catch (error) {
+      console.log(error)
+    }
+  }
+  const formik = useFormik<FormValues>({
```
Comment (Sumick) [inline src/services/edit-profile-form-service.tsx:135]: **nitpick:** Hm, we should not call `useFormik` here if we use `FormikContext` in other place. It's like duplicating the same logic. I think this is one of the issues why you have problems with setting initial values for the edit form

**Review by Sumick — CHANGES_REQUESTED** [review]
There are important 4 things to improve:

- [ ]  The way we fetch profile data, it should have been done server side, not client side using `useProfile` hook
- [ ] We need to specify what are backend types, and frontend types to not mix them
- [ ] The way we initialize formik form. Currently we duplicate this logic when calling `useFormik` and Formik context. It's basically the same but for two different use cases. We need to refactor it
- [ ] We need to set initial values for edit form when above point is refactored

**Pull request merged**